client/my-sites/plugins/plugin-list-header/index.jsx

```diff
@@ -38,31 +39,22 @@ function checkPropsChange( nextProps, propArr ) {
 	return false;
 }
 
-export default React.createClass( {
-	displayName: 'Plugins-list-header',
-
-	propTypes: {
-		label: React.PropTypes.string,
-		isBulkManagementActive: React.PropTypes.bool,
-		toggleBulkManagement: React.PropTypes.func.isRequired,
-		updateAllPlugins: React.PropTypes.func.isRequired,
-		updateSelected: React.PropTypes.func.isRequired,
-		haveUpdatesSelected: React.PropTypes.bool,
-		pluginUpdateCount: React.PropTypes.number.isRequired,
-		activateSelected: React.PropTypes.func.isRequired,
-		deactiveAndDisconnectSelected: React.PropTypes.func.isRequired,
-		deactivateSelected: React.PropTypes.func.isRequired,
-		setAutoupdateSelected: React.PropTypes.func.isRequired,
-		setSelectionState: React.PropTypes.func.isRequired,
-		unsetAutoupdateSelected: React.PropTypes.func.isRequired,
-		removePluginNotice: React.PropTypes.func.isRequired,
-		haveActiveSelected: React.PropTypes.bool,
-		haveInactiveSelected: React.PropTypes.bool,
-		bulkManagement: React.PropTypes.bool,
-		selectedSiteSlug: React.PropTypes.string,
-		plugins: React.PropTypes.array.isRequired,
-		selected: React.PropTypes.array.isRequired
-	},
+export class PluginsListHeader extends Component {
+
+	constructor( props ) {
+		super( props );
+		this.state = {
+			actionBarVisible: _actionBarVisible,
+			addPluginTooltip: false
+		};
+
+		this.toggleBulkManagement = this.toggleBulkManagement.bind( this );
+		this.onBrowserLinkClick = this.onBrowserLinkClick.bind( this );
+		this.afterResize = this.afterResize.bind( this );
+		this.showPluginTooltip = this.showPluginTooltip.bind( this );
+		this.hidePluginTooltip = this.hidePluginTooltip.bind( this );
+		this.unselectOrSelectAll = this.unselectOrSelectAll.bind( this );
+	}
 
 	shouldComponentUpdate( nextProps, nextState ) {
```

Comment:
All we're doing here is props and state checking; would it make sense to use a `React.PureComponent` instead and avoid all of this logic, including `checkPropsChange`?